### PR TITLE
test: Migrate KFP SDK Execution Tests to GHA

### PR DESCRIPTION
**Pull request by DharmitD**
**Description of your changes:**
Resolves https://github.com/kubeflow/pipelines/issues/10925

Converts the KFP SDK Execution test to a GH Action Workflow.

Here's a working GH Action run on my fork: https://github.com/DharmitD/data-science-pipelines-argo/actions/runs/9751594133/job/26913514655

PR to remove the KFP Execution test from prow config: https://github.com/GoogleCloudPlatform/oss-test-infra/pull/2313

**Checklist:**
- [x] The title for your pull request (PR) should follow our title convention. [Learn more about the pull request title convention used in this repository](https://github.com/kubeflow/pipelines/blob/master/CONTRIBUTING.md#pull-request-title-convention).

### CODE REVIEW

**Comment by DharmitD** [comment]
Ready for review.
cc: @hbelmiro @rimolive @chensun

File: .github/workflows/sdk-execution.yml
Comment (hbelmiro) [inline .github/workflows/sdk-execution.yml:32]: You can use the `kfp-cluster` action here. Like https://github.com/kubeflow/pipelines/blob/7d2ec3951594b74043c7c83822536a75fadb1055/.github/workflows/e2e-test.yaml#L27

File: .github/workflows/sdk-execution.yml
Comment (hbelmiro) [inline .github/workflows/sdk-execution.yml:40]: Don't we want to test with the other versions as well?
You can use a matrix for that. See https://github.com/kubeflow/pipelines/pull/10926/files#diff-e842340381d5ec7517ff5c2522b081e6b081cad5701ce9f06af7d088428974eeR17-R24

**Review by hbelmiro — CHANGES_REQUESTED** [review]
/hold

File: .github/workflows/sdk-execution.yml
Comment (DharmitD) [reply .github/workflows/sdk-execution.yml:40]: the Prow Job used only Python 3.8: https://github.com/GoogleCloudPlatform/oss-test-infra/blob/master/prow/prowjobs/kubeflow/pipelines/kubeflow-pipelines-presubmits.yaml#L238

Do we need this for other versions? I would suggest keeping it simple, to one version, and not adding more jobs / more time to execute tests for every PR.

File: .github/workflows/sdk-execution.yml
Comment (DharmitD) [reply .github/workflows/sdk-execution.yml]: Done, updated to use this action.

File: .github/workflows/sdk-execution.yml
Comment (hbelmiro) [reply .github/workflows/sdk-execution.yml:29]: > Do we need this for other versions?
I don't have a strong opinion, but testing with all supported versions would be nice.

> I would suggest keeping it simple, to one version, and not adding more jobs / more time to execute tests for every PR.
The jobs run in parallel, so unless we reach the limit of parallel executions, it shouldn't increase the execution time.

@rimolive any thoughts?

Comment (hbelmiro) [reply .github/workflows/sdk-execution.yml:29]: Only 4 jobs run in parallel. I just saw that on my PR.
So yeah, it would increase the execution time.

Screenshot from 2024-07-02 13-22-38

**Review by hbelmiro — APPROVED** [review]
/unhold
/lgtm

**Comment by rimolive** [comment]
@zijianjoy Can you help us? PR is already approved but because Prow had failed tests it does not merge. If you take a look, the correspondent GHA workflow with the failed Prow test passed with no issues.

**Comment by hbelmiro** [comment]
> @zijianjoy Can you help us? PR is already approved but because Prow had failed tests it does not merge. If you take a look, the correspondent GHA workflow with the failed Prow test passed with no issues.

@rimolive @zijianjoy just to complement, https://github.com/GoogleCloudPlatform/oss-test-infra/pull/2313 removes the `kubeflow-pipelines-sdk-execution-tests` test from Prow. Once it get's merged, we just need to rebase this PR.

**Comment by hbelmiro** [comment]
@james-jwu can you help us to get this PR and https://github.com/GoogleCloudPlatform/oss-test-infra/pull/2313 merged?
The Prow test is failing and blocking other PRs. See: https://github.com/kubeflow/pipelines/pull/10979#issuecomment-2233729432

**Comment by DharmitD** [comment]
> 

> @james-jwu can you help us to get this PR and [GoogleCloudPlatform/oss-test-infra#2313](https://github.com/GoogleCloudPlatform/oss-test-infra/pull/2313) merged? The Prow test is failing and blocking other PRs. See: [#10979 (comment)](https://github.com/kubeflow/pipelines/pull/10979#issuecomment-2233729432)

@james-jwu @chensun could you please take a look and approve?

**Comment by DharmitD** [comment]
This PR is ready for merge. @hbelmiro @rimolive could you please LGTM?

**Review by hbelmiro — COMMENTED** [review]
/lgtm

**Pull request merged**
